src/compiler/checker.ts

```diff
@@ -47511,10 +47511,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     // Fully type check a source file and collect the relevant diagnostics.
-    function checkSourceFileWorker(node: SourceFile) {
+    function checkSourceFileWorker(node: SourceFile, skipCheck: boolean | undefined) {
         const links = getNodeLinks(node);
         if (!(links.flags & NodeCheckFlags.TypeChecked)) {
-            if (skipTypeChecking(node, compilerOptions, host)) {
+            if (skipCheck || skipTypeChecking(node, compilerOptions, host)) {

```

Comment:
Personally, for this one, I'd just modify the implementation of the `skipTypeChecking` function to look for the `ts-nocheck` comment, in addition to its current conditions, rather than passing through all this context.

Reply:
But if i move that into skipTypeChecking, it will disable suggestion diagnostics for files with "ts-noCheck" may be thats ok but just making sure it is intended.

Reply:
Yeah, but being included in this condition would *also* skip generating suggestions, anyway. Which is, imo, the point - you either do the full diagnostic pass and get suggestions or you don't and... don't. Anyway, keeping it all inside that one `skipTypeChecking` function simplifies things when other places need to ask "hey, is this file typechecked?", which should honestly be the condition in `emitter.ts`, rather than things-which-approximate-that.